src/lib/storage/segment_access_counter.hpp

```diff
@@ -10,9 +10,12 @@
 #include "types.hpp"
 
 namespace opossum {
-class Table;
 
-// The SegmentAccessCounter is a collection of counters to register how often and how a segment is accessed.
+// The SegmentAccessCounter is a collection of counters to count how often a segment is accessed.
+// It contains several counters (see AccessType) to differentiate between different access types, like
+// sequential or random access. The individual counters can be accessed using the [] operator.
+// The counters are currently updated by the iterators, segment accessors or from withing the segment itself.
+// Use function access_type(const PosList& positions) to derive the access pattern from the position list.
```

Comment:
Suggestion: Move the last line to the function itself and make it "For a given position list, this determines whether its entries are in sequential, monotonic, or random order. It only looks at the first n values."